rfd/0067-desktop-access-file-system-sharing.md

```diff
@@ -0,0 +1,541 @@
+                       | Client Device List Announce Request|
+                       |<-----------------------------------+
+                       |                                    |
+                       |Server Device Announce Response     |
+                       +----------------------------------->|
+                       |                                    |
+                       |Device Create Request               |Generates a FileId corresponding to
+                       |(IRP_MJ_CREATE)                     |the requested Path (which is "", meaning
+                       +----------------------------------->|the top level directory), internally creating some
+                       |                                    |sort of file handle.
+                       |Device Create Response              |
+                       |<-----------------------------------+and responds with the FileId.
+                       |                                    |
+                       |Drive Query Information Request     |
+Asks for metadata about|(IRP_MJ_QUERY_INFORMATION)          |
+the tld.               +----------------------------------->|
+                       |                                    |
+                       |Drive Query Information Response    |
+                       |<-----------------------------------+Replies with the metadata
+                       |                                    |
+                       |Device Close Request                |
+                       |(IRP_MJ_CLOSE)                      |Operation complete, the previously
+                       +----------------------------------->|generated FileId can now be recycled
+                       |                                    |and internal file handle can be deleted.
+                       |                                    |
+                       |                                    |
+                       |                                    |
+```
+
+A similar process would then take place for other operations, for example a read of a file in the shared directory named `example.txt` would look like
+
+1. server sends `IRP_MJ_CREATE` with `Path: "example.txt"`
+2. client responds with a `FileId`
+3. server sends `IRP_MJ_READ` for `FileId` for bytes 0 - 1024
+4. client executes the read and responds with the data
+5. server sends `IRP_MJ_CLOSE`
+
+## TDP File Shared Directory Extension
+
+Our RDP client lives on the Windows Desktop Service, while the directory we're sharing is exposed to us via the user's browser. This means that in order to get information to and from the shared directory,
+we must extends the TDP protocol (see the diagram in the `Introduction` section for reference).
+
+Each `* Request` and `* Response` TDP message contains a `completion_id` field, with `* Request`s being responsible for generating the `completion_id`s, and `* Response`s being responsible for
```

Comment:
Could you please tell me (and possibly describe it in this section) what is `* Request` or `* Response`? Does it mean `any` request or response? It's a bit unclear to me what does it mean in this context.